### PR TITLE
detect-sameip: convert unittest to FAIL/PASS APIs

### PR DESCRIPTION
**Pull request by TheKharleeci**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/4057

Ticket: #4057

Previous Pr: #6534 

Describe changes:

- Update Detect-sameip unittests to use FAIL/PASS APIs
- Update Copyright Year

### CODE REVIEW

File: src/detect-sameip.c [hunk 1]
```diff
@@ -143,45 +142,33 @@ static int DetectSameipSigTest01(void)
...
     SigGroupBuild(de_ctx);
     DetectEngineThreadCtxInit(&th_v, (void *)de_ctx, (void *)&det_ctx);
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p1);
-    if (PacketAlertCheck(p1, 1) == 0) {
-        printf("sid 2 did not alert, but should have: ");
-        goto cleanup;
-    }
+    FAIL_IF(PacketAlertCheck(p1, 1) == 0);
+    cleanup
```
Comment (jufajardini) [inline src/detect-sameip.c:161]: Hey, I think my previous feedback was misleading. What I meant was that we don't need the `goto` statement any longer, altogether ;)

File: src/detect-sameip.c [hunk 2]
```diff
@@ -143,45 +142,33 @@ static int DetectSameipSigTest01(void)
...
-    if (PacketAlertCheck(p2, 1) != 0) {
-        printf("sid 2 alerted, but should not have: ");
-        goto cleanup;
-    }
-
-    result = 1;
+    FAIL_IF(PacketAlertCheck(p2, 1) != 0);
+    cleanup
 
 cleanup:
```
Comment (jufajardini) [inline src/detect-sameip.c:167]: Because we are not using the `goto`, there is no need to keep this `cleanup:`, either ;)

**Comment by jufajardini** [comment]
These fails are must be checked and addressed, too, please. 

For the formatting failure, you may check https://github.com/OISF/suricata/blob/master/doc/devguide/codebase/code-style.rst
It explains how to install and run clang, so you can check and fix formatting issues.

**Comment by jufajardini** [comment]
When you submit the new PR, can you please add the ticket number from redmine in the body of the commit message? Something like:
```
detect-sameip: convert unittest to FAIL/PASS APIs

Bug: #4057
```

This makes it much easier for everyone to find and things, when working with git-log :)

**Comment by TheKharleeci** [comment]
> When you submit the new PR, can you please add the ticket number from redmine in the body of the commit message? Something like:
> 
> ```
> detect-sameip: convert unittest to FAIL/PASS APIs
> 
> Bug: #4057
> ```
> 
> This makes it much easier for everyone to find and things, when working with git-log :)

Ok. Thank you for the corrections

**Comment by TheKharleeci** [comment]
Followed by: #6558

**Pull request closed without merging**